pkg/config/config.go

```diff
@@ -80,6 +80,34 @@ type Server struct {
 	CacheDictionarySize int               `deprecated:"true"`
 	CacheSegmentSize    int               `deprecated:"true"`
 	CacheTreeSize       int               `deprecated:"true"`
+
+	GoogleEnabled      bool   `def:"false" desc:"enables Google Oauth"`
+	GoogleClientID     string `def:"<yourClientID>" desc:"client ID generated for Google API"`
+	GoogleClientSecret string `def:"<yourClientSecret>" desc:"client secret generated for Google API"`
+	GoogleRedirectURL  string `def:"<pyroscope.mycompany.org/google/callback>" desc:"url that google will redirect to after logging in. Has to be in form <pathToPyroscopeServer/google/callback>"`
```

Comment:
Can we have a way of determining this automatically from request Host header?

Reply:
I think ideally:
* it should still be configurable
* by default just be an empty string
* and if it's an empty string the server just guesses it from host header and current request scheme (http / https)
* and we could add something to `desc` mentioning that you should only set it if pyroscope is under some fancy load balancer

Reply:
Hm interesting idea ... In general Host header shouldn't be trusted as they can be spoofed, but in this case I can't think of a problem that would cause us. Worst case scenario I can think of is spoofing the Host header which leads to wrong redirect URL and "400" error page from google. I'll look into doing this.